enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def my_each
+    each do |value|
+      yield(value)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |value|
+      yield(value, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    each do |value|
+      new_array.push(value) if yield(value)
+    end
+    new_array
+  end
```

Comment:
* [ ] `my_select` is identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_select returns an enumerator if no block is given
  ```
    array = [1,2,5,8,7,98,4]
    p array.my_select.class # not Enumerator
  ```